kie-server-parent/kie-server-services/kie-server-services-dmn/src/main/java/org/kie/server/services/dmn/ModelEvaluatorServiceBase.java

```diff
@@ -204,8 +218,128 @@ public static DMNItemDefinitionInfo itemDefinitionToInfo(ItemDefinition itemDef)
         }
     }
 
+    public Response evaluateModel(String containerId, String modelId, String contextPayload, boolean asDmnResult, String decisionServiceId) {
+        try {
+            KieContainerInstanceImpl kContainer = context.getContainer(containerId, ContainerLocatorProvider.get().getLocator());
+            DMNRuntime dmnRuntime = KieRuntimeFactory.of(kContainer.getKieContainer().getKieBase()).get(DMNRuntime.class);
+
+            List<DMNModel> modelsWithID = dmnRuntime.getModels().stream().filter(m -> m.getName().equals(modelId)).collect(Collectors.toList());
+            if (modelsWithID.isEmpty()) {
+                return Response.status(Status.NOT_FOUND).entity("No model identifies with modelId: " + modelId).build();
+            } else if (modelsWithID.size() > 1) {
+                return Response.status(Status.NOT_FOUND).entity("More than one existing DMN model having modelId: " + modelId).build();
```

Comment:
A nitpick:
Maybe in this case the HTTP 500 would be ok, as more models are found. That is not a problem of request but rather server problem I suppose (not much familiar with DMN).

Reply:
In this case "the server couldn't find which model you actually meant" (since there are homonym models deployed).
Just for consistency, I would keep a 404, this is not really the server broke internally, is indeed the _**missing**_ of univocally identify a model.

~ " I couldn't find which model you meant? "

Reply:
ok, fine for me, thanks